apps/external-contracts-tests/test-suites/national-registry.spec.ts

```diff
@@ -1,24 +1,55 @@
-// NOTE: To run this locally, you'll need to portforward soffia and set
-// the environment variable "SOFFIA_SOAP_URL" to https://localhost:8443
-// kubectl port-forward svc/socat-soffia 8443:443 -n socat
-import { NationalRegistryApi } from '@island.is/clients/national-registry-v1'
+// NOTE: To run this locally, you'll need to portforward xroad service port and set
+// the environment variable "XROAD_BASE_PATH_WITH_ENV=http://localhost:8088/r1/IS-TEST"
+// Example:
+//
+// $ kubectl port-forward svc/socat-xroad 8088:80 -n socat &
+// $ XROAD_BASE_PATH_WITH_ENV=http://localhost:8088/r1/IS-DEV XROAD_CLIENT_ID=IS-DEV/GOV/10000/island-is-client XROAD_TJODSKRA_API_PATH=/SKRA-Protected/Einstaklingar-v1 XROAD_TJODSKRA_MEMBER_CODE=10001 yarn nx run external-contracts-tests:external-test
+
+import { Test, TestingModule } from '@nestjs/testing'
+import {
+  createXRoadAPIPath,
+  XRoadMemberClass,
+} from '@island.is/shared/utils/server'
+
+import {
+  NationalRegistryModule,
+  EinstaklingarApi,
+} from '@island.is/clients/national-registry-v2'
 
 describe('is.island.external.national', () => {
-  let client: NationalRegistryApi
+  let client: EinstaklingarApi
+  const XROAD_CLIENT = process.env.XROAD_CLIENT_ID!
+
   beforeAll(async () => {
-    client = await NationalRegistryApi.instantiateClass({
-      baseSoapUrl: process.env.SOFFIA_SOAP_URL!,
-      user: process.env.SOFFIA_USER!,
-      password: process.env.SOFFIA_PASS!,
-      host: process.env.SOFFIA_HOST_URL!,
-    })
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        NationalRegistryModule.register({
+          xRoadPath: createXRoadAPIPath(
+            process.env.XROAD_BASE_PATH_WITH_ENV!,
+            XRoadMemberClass.GovernmentInstitution,
+            process.env.XROAD_TJODSKRA_MEMBER_CODE!,
+            process.env.XROAD_TJODSKRA_API_PATH!,
+          ),
+          xRoadClient: XROAD_CLIENT,
+        }),
```

Comment:
You need to merge or rebase from main, since all of this has changed to use the ConfigModule.

Same with the environment variable comment above. Shouldn't need any environment variables locally as is (currently, the config has good default values for development).

Reply:
Since this API usage seems like a lot of work that I am not familiar with I will comment this out for future reference and replace with fake success responses. Since this PR is to put the infrastructure in place, I will focus on that.
Alternatively I can revert to using the SOFIA API since that was working and is a bit better than fake tests.

@eirikurn Any preferences on the above?